e2e2/internal/metric/metric.go

```diff
@@ -0,0 +1,192 @@
+package metric
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"time"
+
+	"github.com/aws/aws-k8s-tester/e2e2/internal/utils"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	v4 "github.com/aws/aws-sdk-go-v2/aws/signer/v4"
+	"github.com/aws/aws-sdk-go-v2/credentials/stscreds"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/prometheus/prompb"
+)
+
+type MetricManager struct {
+	Metadata              map[string]string
+	AwsConfig             aws.Config
+	AmpRemoteWriteURL     string
+	AmpRemoteWriteRoleArn string
+}
+
+func NewMetricManager(metadata map[string]string, awsConfig aws.Config, url string, role string) *MetricManager {
+
+	return &MetricManager{
+		Metadata:              metadata,
+		AwsConfig:             awsConfig,
+		AmpRemoteWriteURL:     url,
+		AmpRemoteWriteRoleArn: role,
+	}
+}
+
+// PushMetricsToAMP pushes metric data to AWS Managed Prometheus (AMP) using SigV4 authentication
+func (m *MetricManager) PushMetricsToAMP(name string, help string, value float64) error {
```

Comment:
1. batching the samples would be preferable to making a separate call to the remote_write API for every sample we collect, IMO

2. are you not able to use the upstream remote write client because of the assume-role jump? https://github.com/prometheus/prometheus/blob/5037cf75f2d4f1671ad365ba1e99902fc36808d5/storage/remote/client.go#L180

Reply:
For the first point, that sounds good—I’ll change it in the next revision. As for the second point, I spent some time trying to use the remote write client, but I wasn’t able to integrate it into my code.